src/Apps/PartnerOffer/Routes/PartnerOfferCheckout.tsx

```diff
@@ -0,0 +1,66 @@
+import { FC, useCallback, useEffect } from "react"
+import { useSystemContext } from "System/SystemContext"
+import { usePartnerOfferCheckoutMutation } from "./Mutations/UsePartnerOfferCheckoutMutation"
+import { useRouter } from "System/Router/useRouter"
+import { LoadingArea } from "Components/LoadingArea"
+import { Box } from "@artsy/palette"
+
+export const PartnerOfferCheckout: FC = () => {
+  const { router } = useSystemContext()
+  const { match } = useRouter()
+  const partnerOfferId = match.params.partnerOfferID
+  const {
+    submitMutation: partnerCheckoutMutation,
+  } = usePartnerOfferCheckoutMutation()
+
+  const handleRedirect = useCallback(async () => {
+    if (partnerOfferId) {
+      try {
+        const response = await partnerCheckoutMutation({
+          variables: {
+            input: {
+              partnerOfferId: partnerOfferId,
+            },
+          },
+        })
+
+        let redirectUrl = "/"
+        let orderOrError = response.commerceCreatePartnerOfferOrder?.orderOrError
+
+        if (orderOrError?.error) {
+          const errorCode = orderOrError.error.code
+          const errorData = JSON.parse(orderOrError.error.data?.toString() ?? "")
+
+          switch (errorCode) {
+            case "expired_partner_offer":
+              // TODO: Show error message in artwork page
+              redirectUrl = `/artwork/${errorData.artwork_id}`
+              break
+            case "not_acquireable":
+              // TODO: Show error message in artwork page
+              redirectUrl = `/artwork/${errorData.artwork_id}`
+              break
+          }
+        } else {
+          redirectUrl = `/orders/${orderOrError?.order?.internalID}`
+        }
+        router?.push(redirectUrl)
+      } catch (error) {
+        console.error("Error creating partner offer order", error)
+      }
+    }
+  }, [partnerCheckoutMutation, partnerOfferId, router])
+
+  useEffect(() => {
+    if (router) {
+      handleRedirect()
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [router])
```

Comment:
@leamotta - a couple things with this code: 
1. we should use the router from `const { router } = useRouter()`. We should be using the system router, not the one that imports from `found` 
2. this code effect code should actually have an empty dependency array, since you want to be sure -- absolutely sure -- that it only fires once. That router reference as used here could change. An empty array ensures something only fires once, on mount.

Reply:
Thanks @damassi! Now I see why the router was nil on mount, leaving me to add it to the dependency array for the logic to work.
I've addressed this in https://github.com/artsy/force/pull/13230